lib/seahorn/HornifyModule.cc

```diff
@@ -142,6 +169,19 @@ bool HornifyModule::runOnModule(Module &M) {
   if (Step == hm_detail::CLP_SMALL_STEP ||
       Step == hm_detail::CLP_FLAT_SMALL_STEP)
     m_sem.reset(new ClpOpSem(m_efac, *this, M.getDataLayout(), TL));
+  else if (InterProcMem) {
+    ShadowMemPass * smp = getAnalysisIfAvailable<sea_dsa::ShadowMemPass>();
+    assert(smp);
+    ShadowMem &shadowmem_analysis =smp->getShadowMem();
+
+    CompleteCallGraph *ccg = getAnalysisIfAvailable<sea_dsa::CompleteCallGraph>();
+    std::shared_ptr<InterMemPreProc> preproc(new InterMemPreProc(ccg, shadowmem_analysis));
```

Comment:
use `std::make_shared`

Reply:
or maybe `std::unique`?

Reply:
I had to make a shared pointer because the `LegacyOperationalSemantics` has a constructor: 
```
  explicit LegacyOperationalSemantics(const LegacyOperationalSemantics &o)
      : OperationalSemantics(o){};
```

Reply:
This does not seem to be a field. At any rate, a comment can be added to address the choice of `shared_ptr` in here.